test/functional/api/vim_spec.lua

```diff
@@ -2878,6 +2892,111 @@ describe('API', function()
+        setlocal  signcolumn=yes        "  8, window-local
+        setlocal  formatprg=myprg       "  9, global-local (buffer)
+        setlocal  virtualedit=block     " 10, global-local (window)
+      ]])
+
+      exec_lua 'vim.cmd.vsplit()'
+      meths.create_buf(false, false)
+
+      bufs = meths.list_bufs()
+      wins = meths.list_wins()
+
+      meths.win_set_buf(wins[1].id, bufs[1].id)
+      meths.win_set_buf(wins[2].id, bufs[2].id)
+
+      meths.set_current_win(wins[2].id)
+      meths.exec('source ' .. fname, false)
+
+      meths.set_current_win(wins[1].id)
+    end)
+
+    after_each(function()
+      os.remove(fname)
+    end)
+
+    it('should return option information', function()
+      eq(meths.get_option_info('completeopt'),    meths.get_option_info_ex('completeopt',   {}))
+      eq(meths.get_option_info('completefunc'),   meths.get_option_info_ex('completefunc',  {}))
+      eq(meths.get_option_info('concealcursor'),  meths.get_option_info_ex('concealcursor', {}))
+      eq(meths.get_option_info('dictionary'),     meths.get_option_info_ex('dictionary',    {}))
+      eq(meths.get_option_info('showbreak'),      meths.get_option_info_ex('showbreak',     {}))
+      eq(meths.get_option_info('wildoptions'),    meths.get_option_info_ex('wildoptions',   {}))
+      eq(meths.get_option_info('spelloptions'),   meths.get_option_info_ex('spelloptions',  {}))
+      eq(meths.get_option_info('signcolumn'),     meths.get_option_info_ex('signcolumn',    {}))
+      eq(meths.get_option_info('formatprg'),      meths.get_option_info_ex('formatprg',     {}))
+      eq(meths.get_option_info('virtualedit'),    meths.get_option_info_ex('virtualedit',   {}))
+    end)
+
+    describe('last set', function()
+      it('should point to the global value if set', function()
+        local info = meths.get_option_info_ex('completefunc', {scope = 'global'})
+        eq(2, info.last_set_linenr)
+        eq(1, info.last_set_sid)
+      end)
+
+      it('should fall back to the global value if buffer local value is not set', function()
+        local info = meths.get_option_info_ex('completefunc', {buf = bufs[2].id})
+        eq(2, info.last_set_linenr)
+        eq(1, info.last_set_sid)
+      end)
+
+      it('should point to the buffer local value if set', function()
+        local info = meths.get_option_info_ex('spelloptions', {buf = bufs[2].id})
+        eq(7, info.last_set_linenr)
+        eq(1, info.last_set_sid)
+      end)
+
+      it('should default to the local value for the current buffer', function()
+        local info = meths.get_option_info_ex('spelloptions', {})
+        eq(0, info.last_set_linenr)
+        eq(0, info.last_set_sid)
+        meths.set_current_win(wins[2].id)
+        info = meths.get_option_info_ex('spelloptions', {})
+        eq(7, info.last_set_linenr)
+        eq(1, info.last_set_sid)
+      end)
+
+      it('should fall back to the global value if window local value is not set', function()
```

Comment:
nit: "should" is never needed in the test names. These names are getting really long so might as well drop it.
```suggestion
      it('falls back to global value if window-local value not set', function()
```